rules/azure/initial_access_external_guest_user_invite.toml

```diff
@@ -0,0 +1,52 @@
+[metadata]
+creation_date = "2020/08/31"
+ecs_version = ["1.6.0"]
+maturity = "production"
+updated_date = "2020/08/31"
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies an invitation to an external user in Azure Active Directory (AD). Azure AD is extended to include
+collaboration, allowing you to invite people from outside your organization to be guest users in your cloud account.
+Unless there is a business need to provide guest access to any user, avoid creating guest users. Guest users could
+potentially be overlooked indefinitely leading to a potential vulnerability.
+"""
+false_positives = [
+    """
+    Guest user invitations may be sent out by a system or network administrator. Verify whether the username, hostname,
+    and/or resource name should be making changes in your environment. Guest user invitations from unfamiliar users or
+    hosts should be investigated. If known behavior is causing false positives, it can be exempted from the rule.
+    """,
+]
+from = "now-25m"
+index = ["filebeat-*"]
+language = "kuery"
+license = "Elastic License"
+name = "Azure External Guest User Invitation"
+note = "The Azure Filebeat module must be enabled to use this rule."
+references = ["https://docs.microsoft.com/en-us/azure/governance/policy/samples/cis-azure-1-1-0"]
+risk_score = 21
+rule_id = "141e9b3a-ff37-4756-989d-05d7cbf35b0e"
+severity = "low"
+tags = ["Elastic", "Azure", "SecOps", "Continuous Monitoring", "Identity and Access"]
+type = "query"
+
+query = '''
+event.module:azure and event.dataset:azure.auditlogs and event.category:AuditLogs and azure.auditlogs.operation_name:"Invite external user" and azure.auditlogs.properties.target_resources.*.display_name:guest and event.outcome:Success
```

Comment:
Is the `*` required in `azure.auditlogs.properties.target_resources.*.display_name:guest`?

Reply:
it's not required, but I wasn't sure if that would ever change. currently it has a 0 -  `azure.auditlogs.properties.target_resources.0.display_name`.  we discussed with another rule and it works with ES and passes tests, so we should be good, its not at the beginning or end. this is also going to crop up with GCP rules as many use versions, like v1 or v2, and we'll probably need to make the query `v*`

Reply:
Yea I remember the convo but not the outcome. Sounds good to me - we will just need to keep an eye on the module too I suppose